python/gym_ignition/rbd/idyntree/inverse_kinematics_nlp.py

```diff
@@ -257,7 +268,7 @@ def add_target(
         elif target_type == TargetType.POSE:
             # Add the target
             ok_target = self._ik.addTarget(
-                frame_name, idt.Transform.Identity(), weight, weight
+                frame_name, idt.Transform.Identity(), weight[0], weight[1]
```

Comment:
If `weight_type`  is float, what is `weight[1]`  returning?

Reply:
There's a check above for the type of the argument that raises an exception in the case the user passes the wrong type

Reply:
Ah, I misread that then. If we don't support scalar anymore, I don't know if it could make sense to document this change somewhere?

Reply:
The weight could either be a float for POSITION and ROTATION targets, or a tuple for POSE target. Right now this class has no explicit documentation but errors are quite descriptive, I need to find some time for both this class and KinDynComputations :)

Reply:
> The weight could either be a float for POSITION and ROTATION targets, or a tuple for POSE target. Right now this class has no explicit documentation but errors are quite descriptive, I need to find some time for both this class and KinDynComputations :)

Ack, even if I would not conflate the topic of documenting a complete class (that in my experience if it is not there at the beginning almost never happens, as it is kind of a huge task : ) ) with documenting a breaking change, that instead is something that can be easily done with a single phrase in a simple CHANGELOG.md or simialr document, and can be easily done in the PR that makes the change itself.

Reply:
A good point that was missing is that previous code is not compatible with this change. I addressed backward compatibility in e1626cf.

Reply:
I just saw the backward compatibility commit, so I guess there is no problem!

Reply:
We commented in the same time :) I agree on documenting breaking changes, and we keep them in the release changelog (trying to minimize them). I realized you were talking about backward compatibility after few comments, I totally agree with https://github.com/robotology/gym-ignition/pull/388#discussion_r709024125 when it happens.